### DISCUSSION TITLE
Filter search by custom fields

### DISCUSSION THREAD

**Discussion by nzdev** [umbraco/Umbraco-CMS#11302] [Features and ideas]
With thousands of child items of a node the list view and any picker datatypes become very challenging to use. It would be super useful to be able to assign custom fields to search by to a list view and picker datatype. Something that ends up looking like the search in Diplo Log Viewer as an example. 
For the content I'm looking at it doesn't make sense to structure the content in a tree.
I've seen a blog where you can intercept the http call to search if you are searching by text only, however I am wanting to search by other field types (drop down lists, date ranges, facets).

Happy to contribute code to this if I can get some guidance.

Thanks

**Comment by patrickdemooij9** [comment]
I agree with this. We've now had multiple projects where we would be overriding the default search behaviour of Umbraco in a listview and replacing it with something of ourselves (https://dev.to/skttl/how-to-customize-searching-in-umbraco-list-views-1knk. This is probably the blog post that you are referring to).

It would be nice if we had an official way to intercepting the listview search behaviour instead of using interceptors for it. Probably being able to register a search implementation for a specific listview.
We can then also take a look at extending that to the tags that you are talking about. Could be part of the search implementation for a specific listview through a special method. Though, I am sure that will probably require some more work.
